lib/options-handler.js

```diff
@@ -277,7 +282,55 @@ export class ClientOptionsHandler {
         return '9999999.99999.999';
     }
 
-    setCompilers(compilers) {
+    getRemoteId(remoteUrl, language) {
+        const url = new URL(remoteUrl);
+        return url.host.replace(/\./g, '_') + '_' + language;
+    }
+    
+    libArrayToObject(libsArr) {
+        const libs = {};
+        for (const lib of libsArr) {
+            libs[lib.id] = lib;
+
+            const versions = lib.versions;
+            lib.versions = {};
+
+            for (const version of versions) {
+                lib.versions[version.id] = version;
+            }
+        }
+        return libs;
+    }
+
+    async getRemoteLibraries(language, remoteUrl) {
+        const remoteId = this.getRemoteId(remoteUrl, language);
+        if (!this.remoteLibs[remoteId]) {
+            return new Promise((resolve) => {
+                const url = remoteUrl + '/api/libraries/' + language;
+                logger.info(`Fetching remote libraries from ${url}`);
+                let fullData = '';
+                https.get(url, (res) => {
```

Comment:
We should find an easier way to do this (not in this PR!) but surely there's some node.js async-able library to fetch JSON data :)

Reply:
With a bit of luck, for the next LTS we get `https://github.com/nodejs/node/pull/41749` not in experimental mode and can just use `fetch` :)